packages/ton-client/src/main.ts

```diff
@@ -268,3 +512,63 @@ function makeDrawScreenQuad(gl: WebGL2RenderingContext) {
   }
 }
 
+function makeDrawCube(gl: WebGL2RenderingContext) {
+  /*
+   * ScreenQuad render
+   */
+  const cubeVao = gl.createVertexArray()
+  gl.bindVertexArray(cubeVao)
+  gl.enableVertexAttribArray(0)
+
+  const cubeBuffer = gl.createBuffer()
+  gl.bindBuffer(gl.ARRAY_BUFFER, cubeBuffer)
+  {
+    const vertices = [
```

Comment:
what is this??? 😅

Reply:
Thehe, this my friend, is a cube!
6 faces, each consisting of 2 triangles, each consisting of 3 vertices, each consiting of 3 values (x,y,z). 
![image](https://user-images.githubusercontent.com/5720176/218293072-6ec8df5f-6213-4f44-a979-8cc7b1b33118.png)

I use it for rendering our object "inside of it". 
So instead of for the whole screen, I only do raymarching for the pixels of a cube, where you would usually compute texture values or so:
![image](https://user-images.githubusercontent.com/5720176/218293222-23099763-023a-47d8-8330-f14407683c05.png)

Reply:
![Screenshot_2023-02-12-10-01-44-434-edit_org.mozilla.firefox.jpg](https://user-images.githubusercontent.com/5720176/218302041-1a9ef7c9-3226-406b-afc3-10399edbde47.jpg)

Reply:
wow nice, thanks for the explanation